spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -291,35 +287,13 @@ private Map<String, String> getMapWithPrefixedKeys(Map<String, String> map, Stri
 
 	@Override
 	public List<String> getServices() {
-		String spelExpression = this.properties.filter();
-		Predicate<Service> filteredServices;
-		if (spelExpression == null || spelExpression.isEmpty()) {
-			filteredServices = (Service instance) -> true;
-		}
-		else {
-			Expression filterExpr = this.parser.parseExpression(spelExpression);
-			filteredServices = (Service instance) -> {
-				Boolean include = filterExpr.getValue(this.evalCtxt, instance, Boolean.class);
-				if (include == null) {
-					return false;
-				}
-				return include;
-			};
-		}
-		return getServices(filteredServices);
+		return adapter.apply(client).stream().map(s -> s.getMetadata().getName()).toList();
 	}
 
+	@Deprecated(forRemoval = true)
 	public List<String> getServices(Predicate<Service> filter) {
-		if (properties.namespaces().isEmpty()) {
```

Comment:
this method should have never been `public`, as it is called from within this class only. But it's here and we can only drop it in the next major release, as such I need to "dance" a little. `Fabric8DiscoveryServicesAdapter` has to take a `Predicate<Service>` in, though if this method was `private` that would have not been needed. Because of this, I also need to adapt some tests, by passing `null`

Reply:
At the same time, I have a PR that drops the deprecated method and adjusted the code because of that. For example `Fabric8DiscoveryServicesAdapter` accepts a `Predicate<Service> filter` as a constructor parameter, though this is only needed because of the above public method. That PR takes care to properly clean things, not simply delete this public method, but to properly adjust the code everywhere because of that.